tests/test_Subfunction_Module.py

```diff
@@ -0,0 +1,320 @@
+import numpy as np  # pragma: no cover
+import pytest  # pragma: no cover
+from scipy.signal.windows import triang  # pragma: no cover
+
+from Subfunction_Module import (
+    buscamin,
+    modmax,
+    picant,
+    picpost,
+    searchoff,
+    searchon,
+    zerocros,
+)  # pragma: no cover
+
+
+@pytest.mark.buscamin_testmarker
+def test_buscamin():
+    """
+    Test for the "buscamin" function;
+    Located in the Subfunction Module of py_pecg.
+    """
+
+    # Valid Cases
+    Test_Signal_1 = np.zeros(10, dtype=int)
+    assert buscamin(Test_Signal_1) == 0
+    Test_Signal_2 = np.linspace(1, 10, 10)
+    assert (
+        buscamin(Test_Signal_2) == 0
+    )  # No local minimum, only global minimum
+    Test_Signal_3 = np.linspace(10, 1, 10)
+    assert (
+        buscamin(Test_Signal_3) == 0
+    )  # No local minimum, only global minimum
+    Test_Signal_4 = (np.linspace(-5, 5, 11)) ** 2
+    assert buscamin(Test_Signal_4) == 4  # Parabolic Signal
+
+    # Invalid Cases
+    with pytest.raises(ValueError):
+        buscamin(np.ones([2, 2]))
+    with pytest.raises(ValueError):
+        buscamin(np.array([]))
+    with pytest.raises(ValueError):
+        buscamin(np.zeros((2, 3, 4)))
+    return
+
+
+@pytest.mark.searchon_testmarker
+def test_searchon():
+    """
+    Test for the "searchon" function;
+    Located in the Subfunction Module of py_pecg.
+    """
+
+    #                              ___  /\  ___
+    # Modified Triangular Pulse:      \/  \/
+    Test_Signal_1 = np.concatenate(
+        (
+            np.ones(10),
+            1 - 0.5 * triang(10, sym=True),
+            1 + 1 * triang(10, sym=True),
+            1 - 0.5 * triang(10, sym=True),
+            np.ones(10),
+        )
+    )
```

Comment:
Repeats between tests, let's make it a [pytest fixture](https://docs.pytest.org/en/6.2.x/fixture.html)